src/block/Campfire.php

```diff
@@ -0,0 +1,193 @@
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\tile\Campfire as TileCampfire;
+use pocketmine\block\utils\HorizontalFacingTrait;
+use pocketmine\crafting\FurnaceRecipe;
+use pocketmine\entity\Entity;
+use pocketmine\entity\Living;
+use pocketmine\entity\projectile\SplashPotion;
+use pocketmine\event\entity\EntityDamageByBlockEvent;
+use pocketmine\event\entity\EntityDamageEvent;
+use pocketmine\item\FlintSteel;
+use pocketmine\item\Item;
+use pocketmine\item\ItemFactory;
+use pocketmine\item\PotionType;
+use pocketmine\item\Shovel;
+use pocketmine\item\VanillaItems;
+use pocketmine\math\Vector3;
+use pocketmine\math\Facing;
+use pocketmine\player\Player;
+use pocketmine\world\BlockTransaction;
+use pocketmine\world\sound\FireExtinguishSound;
+use pocketmine\world\sound\FlintSteelSound;
+use pocketmine\world\sound\ItemFrameAddItemSound;
+use function count;
+
+class Campfire extends Transparent{
+	use HorizontalFacingTrait;
+
+	protected bool $extinguished = false;
+
+	public function writeStateToMeta() : int{
+		return $this->facing | ($this->extinguished ? BlockLegacyMetadata::CAMPFIRE_FLAG_EXTINGUISHED : 0);
+	}
+
+	public function readStateFromData(int $id, int $stateMeta) : void{
+		$this->facing = $stateMeta & 0x03;
+		$this->extinguished = ($stateMeta & BlockLegacyMetadata::CAMPFIRE_FLAG_EXTINGUISHED) === BlockLegacyMetadata::CAMPFIRE_FLAG_EXTINGUISHED;
+	}
+
+	public function getStateBitmask() : int{
+		return 0b111;
+	}
+
+	public function hasEntityCollision() : bool{
+		return true;
+	}
+
+	public function getLightLevel() : int{
+		return $this->extinguished ? 0 : 15;
+	}
+
+	public function isAffectedBySilkTouch() : bool{
+		return true;
+	}
+
+	public function getDropsForCompatibleTool(Item $item) : array{
+		return [
+			VanillaItems::CHARCOAL()->setCount(2)
+		];
+	}
+
+	public function isExtinguished() : bool{
+		return $this->extinguished;
+	}
+
+	/** @return $this */
+	public function setExtinguished(bool $extinguish = true) : self{
+		$this->extinguished = $extinguish;
+		return $this;
+	}
+
+	private function extinguish() : void{
+		$this->position->getWorld()->addSound($this->position, new FireExtinguishSound());
+		$this->position->getWorld()->setBlock($this->position, $this->setExtinguished());
+	}
+
+	private function fire() : void{
+		$this->position->getWorld()->addSound($this->position, new FlintSteelSound());
+		$this->position->getWorld()->setBlock($this->position, $this->setExtinguished(false));
+		$this->position->getWorld()->scheduleDelayedBlockUpdate($this->position, 1);
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if($this->getSide(Facing::DOWN)->isTransparent()){
+			return false;
+		}
+		if($player !== null){
+			$faces = [
+				4 => 3,
+				3 => 2,
+				2 => 0,
+				5 => 1
+			];
```

Comment:
That's not how this is done. The facing field should contain `Facing::` values only, not magic metadata garbage.